gammapy/image/maps.py

```diff
@@ -291,6 +291,48 @@ def contains(self, position):
         x, y = skycoord_to_pixel(position, self.wcs, self.wcs_origin)
         return (x >= 0.5) & (x <= nx + 0.5) & (y >= 0.5) & (y <= ny + 0.5)
 
+
+    def downsample(self, factor, method=np.nansum, shape=None):
+        """
+        Down sample sky map by a power of two.
+
+        The sky map is down sampled using `skimage.measure.block_reduce`. Only
+        down sampling factor, that are a power of two are allowed. The sky map is
+        padded to a given size using the 'reflect' method, before the down sampling
+        is done.
+
+        Parameters
+        ----------
+        factor : int
+            Down sampling factor, must be power of two.
+        method : np.ufunc (np.nansum), optional
+            Method how to combine the image blocks.
+        shape : tuple (None), optional
+            If shape is specified, the image is padded prior to the down sampling
+            symmetrically in x and y direction to the given shape.
+
+        Returns
+        -------
+        skymap : `~gammapy.image.SkyMap`
+            Down sampled sky map.
+        """
+        from skimage.measure import block_reduce
+        if not np.log2(factor).is_integer():
+            raise ValueError('Downsampling factor must be power of 2.')
+        factor = int(factor)
+
+        if shape is not None:
+            x_pad = (shape[1] - self.data.shape[1]) // 2
+            y_pad = (shape[0] - self.data.shape[0]) // 2
+            #converting from unicode to ascii string as a workaround
+            #for https://github.com/numpy/numpy/issues/7112
+            data = np.pad(self.data, ((y_pad, y_pad), (x_pad, x_pad)), mode=str('reflect'))
+        else:
+            data = self.data
+        new_data = block_reduce(data, (factor, factor), method)
+        return SkyMap(data=new_data)
```

Comment:
I think downsampling actually requires to update the WCS information in the header and creating a new `WCS` object, because the size of the pixel changed. In detail I think this is rather complicated, because the shape of the data changes as well because of padding etc. @cdeil Any idea how to handle this uniformly? Maybe one could achieve the same by just calling `.reproject()`?